docker-compose.yml

```diff
@@ -66,15 +66,23 @@ services:
     # we can wait for services to be up and running
     healthcheck:
       test: ["CMD-SHELL", "DJANGO_SETTINGS_MODULE=olympia celery -A olympia.amo.celery status"]
-      interval: 1s
-      retries: 100
+      # The interval is 90s after the start period of 60s
+      interval: 90s
+      # 3 failed attempts result in container failure
+      retries: 3
+      # While starting, ping faster to get the container healthy as soon as possible
+      start_interval: 1s
+      # The start period is 60s
+      start_period: 60s
```

Comment:
Am I understanding this correctly that if start up takes longer than 60s the next healthcheck is at 2.5 minutes after start up?  What's the downside to having start_period be a longer time?

Reply:
The startup period defines how long to wait before "counting" towards the total retries. If the service starts before that time it will switch to normal healtcheck. 

So if startup takes longer than 60s the next healtcheck is indeed 2.5 after and will try 3 times. BUT. It is unfathomable that startup takes longer than 60s and you don't have a terminally ill container.

It will always be less than that which is why I set it there. Making it longer won't materially change anything.

Additionally you _want_ the normal interval to be long otherwise you get spammed with /__version__ requests in the container logs.

Reply:
We might have to revisit that "always less than 60s" after Andreas and I use it 😿

Reply:
Would 120s be more realistic? give me a number and I'll bump it up.

Reply:
I don't know yet, but we've consistently seen longer times.